Initial Changes to README

tapankavasthi opened the pull request:
- A bit of formatting here, and there
- A few links in and around
- Gave it a few SEO coins
- Gave it a poetic conclusion

Kudos, Sherub! Good to know about this lecture series.

[Preview Changes](https://github.com/tapankavasthi/learning-haskell)

tapankavasthi committed:
Initial Changes to README

- A bit of formatting here, and there
- A few links in and around
- Gave it a few SEO coins
- Gave it a poetic conclusion

Kudos, Sherub! Good to know about this lecture series.

sherubthakur commented on README.md:
````diff
@@ -1,19 +1,75 @@
+$ curl -sSL https://get.haskellstack.org/ | sh
+```
+
+Well, there's a good reason for using the _-L_ flag with our [cURL](http://www.mit.edu/afs.new/sipb/user/ssen/src/curl-7.11.1/docs/curl.html) here. That's because the URL used here responds with a 302 (redirection) HTTP status code, and we need to follow through the redirection.
+
+To dive deeper into this concept, we can look at the header information by using the -D flag:
+
+```shell script
+$ curl -is -o /dev/null -D - https://get.haskellstack.org/
+
+HTTP/2 302
+server: nginx/1.17.8
+date: Mon, 20 Apr 2020 15:55:07 GMT
+content-type: text/html
+content-length: 145
+location: https://raw.githubusercontent.com/commercialhaskell/stack/stable/etc/scripts/get-stack.sh
+```
+
+With the header information available to us, We can now see the actual location of our script.
````
Can we shift this part into FAQ. I would like the readme to be like a no-brainer thing first and foremost. And if folks are interested in knowing more let's address this using an FAQ section.

tapankavasthi replied:
@sherubthakur  Done 🙂

sherubthakur commented on README.md:
````diff
@@ -1,19 +1,75 @@
+
+```shell script
+$ cd learning-haskell
+$ stack build --test
````
Opinion: `stack test` should do everything for you and is less verbose.

tapankavasthi replied:
Yes, both are synonyms - yes?
I've changed it to `stack test` as per your suggestion.

nixypanda reviewed (commented):
This looks great man. I will merge this into the main branch as soon as we move the flag section to FAQ section. Thanks!

tapankavasthi committed:
Include Sherub's Review Suggestions

tapankavasthi commented:
> This looks great man. I will merge this into the main branch as soon as we move the flag section to FAQ section. Thanks!

Thanks, @sherubthakur for taking a look. I've addressed your review comments. Please take a second look at your convenience.
Btw, the idea of including a FAQ guide was spot on!
